src/pip/_internal/req/constructors.py

```diff
@@ -293,6 +294,7 @@ def install_req_from_req(
             "%s depends on %s " % (comes_from.name, req)
         )
 
+    # TODO: use_pep517?
```

Comment:
Good catch, I'll look into that. I agree that just passing the flag through the resolver is the right "for now" solution. Refactoring to reduce the number of parameters that are being passed all over the place would be a *very* good quality of life improvement somewhere down the line, but it's not one I have the energy for right now.

Also, it's worth remembering that `--use-pep517` is only a temporary thing anyway. The intention is that once the legacy code path is no longer needed, the flag and the old code path will go away. I don't have a feel for the timescale on that, but I'm reluctant to invest much effort into maintaining a code path that we want to lose - and personally, I'm in favour of rapidly removing the legacy install code (I just suspect that the backward compatibility demons might fight me on that 😉 ).

Reply:
I do agree that it's a temporary thing but it might be necessary to hold it for at least about an year.

Plus separating the two codepaths in a much better way would probably make life easier moving forward anyway (_wink_ #5051 and more _wink_).

Reply:
... and here was me assuming that #5051 was when you would fix all the mess I'd made of the requirement code, saving me from having to write decent code in the first place 😉

Reply:
@pradyunsg My mistake, the code that uses `--use-pep517` is in `base_command.py`, so that's all good. And the wheel command had the option, it just needed the same fix to pass the value to the resolver. These final stages have just been delayed so long I'm forgetting what's done and what isn't.